core/common/src/main/java/alluxio/conf/InstancedConfiguration.java

```diff
@@ -336,14 +337,31 @@ public void validate() {
     if (!getBoolean(PropertyKey.CONF_VALIDATION_ENABLED)) {
       return;
     }
+    List<String> removalMessages = new ArrayList<>();
     for (PropertyKey key : keySet()) {
       Preconditions.checkState(
           getSource(key).getType() != Source.Type.SITE_PROPERTY || !key.isIgnoredSiteProperty(),
           "%s is not accepted in alluxio-site.properties, "
               + "and must be specified as a JVM property. "
               + "If no JVM property is present, Alluxio will use default value '%s'.",
           key.getName(), key.getDefaultValue());
+
+      if (PropertyKey.isDeprecated(key) && getSource(key).compareTo(Source.DEFAULT) > 0) {
+        LOG.warn("{} is deprecated. Please avoid using this key in the future. {}", key.getName(),
```

Comment:
splitting the warning across multiple files like this makes it harder to generate user-friendly messages. I'd prefer to duplicate the deprecation statement, but will leave it up to you

Reply:
Sorry, I don't quite understand what you mean here? All removal or deprecation messages will always go to the logs. I don't see how it goes to multiple files?

In the case where someone provides a "removed" key we throw the exception and tell them they can't use the key. The exception message provides the offending keys. I log them anyways as an error for good measure though as someone might do some parsing on the log format and it can be useful to see the "ERROR" lines.

Reply:
I mean source code files. Part of the message is here, and part is on the property key. The person writing the property key side needs to know about this side to understand what has already been said